### PR TITLE
Biopete/big data import

### PR DESCRIPTION
**Pull request by biopete**
Added new goal for scripting the loading of big data locally
run-local-bigdata-migrations 

I removed the update concept table sql from liquibase and instead made a new concept.csv with the counts and prevalence already populated.   The big data csvs are in google storage. 

The indexes are dropped on the big tables in liquibase , then tables imported with mysql commands, then added back. The importing of the whole process takes just a few minutes now instead of 30 minutes.  The longest part is adding back the indexes. 

TODO:  Share the index creation code with changelog-schema so indexes are not defined in two places .  

TODO: big data migrations on server 

TODO: add this goal to the run-local-all-migrations

### CODE REVIEW

File: api/db-cdr/run-bigdata-migrations.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+# Add data files to import here
+DATA_FILES=(concept.csv concept_relationship.csv)
+
+# Download data and truncate tables before dropping indexes
+for f in "${DATA_FILES[@]}"
+do
+  local_fpath=/tmp/$f
+  curl -o $local_fpath "$REMOTE_DATA_LOC/$f"
+  #gsutil cp $REMOTE_DATA_BUCKET/$f $f  # todo maybe , gives error now
```
Comment (danqrodney) [inline api/db-cdr/run-bigdata-migrations.sh]: Did you run gcloud activate-service-account or gcloud auth login before trying to do this?

We have Ruby commands which copy stuff from GCS without issue:

https://github.com/all-of-us/workbench/blob/master/api/libproject/devstart.rb#L175

Do those work for you?

File: api/db-cdr/run-bigdata-migrations.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+echo "Dropping indexes on big tables to speed up insert (dramatically) "
+../gradlew update $drop_index_activity $context
+
+# Import the data
+for f in "${DATA_FILES[@]}"
+do
+  local_fpath=/tmp/$f
+  db_name=cdr
+  table_name="${f%\.csv*}"
+  mysqlimport --ignore-lines=1 --fields-terminated-by=, --fields-enclosed-by='"' --verbose --local -h ${DB_HOST} --port ${DB_PORT} -u root -p${MYSQL_ROOT_PASSWORD} cdr $local_fpath
```
Comment (danqrodney) [inline api/db-cdr/run-bigdata-migrations.sh]: Add a \ and wrap this long line?

File: api/db-cdr/run-bigdata-migrations.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+CREATE_DB_FILE=/tmp/create_db.sql
+
+function finish {
+  rm -f ${CREATE_DB_FILE}
+}
+trap finish EXIT
+
+cat create_db.sql | envsubst > $CREATE_DB_FILE
+echo "Creating database if it does not exist..."
+mysql -h ${DB_HOST} --port ${DB_PORT} -u root -p${MYSQL_ROOT_PASSWORD} < ${CREATE_DB_FILE}
```
[the remark this answers is not in the document]
Comment (biopete) [reply api/db-cdr/run-bigdata-migrations.sh]: you are right

**Comment by biopete** [comment]
So i have some changes in but can't test yet because my internet is down from a storm. Thanks for review

**Comment by biopete** [comment]
Oops. Guess is should have just hit the comment button and not close and comment.

File: api/db-cdr/run-bigdata-migrations.sh [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+# Download data and import
+REMOTE_DATA_LOC=https://storage.googleapis.com/all-of-us-ehr-dev-peter-speltz
```
[the remark this answers is not in the document]
Comment (biopete) [reply api/db-cdr/run-bigdata-migrations.sh]: yeah. I was just trying out the gsutil cp. One or the other whichever way.

File: api/db-cdr/run-bigdata-migrations.sh [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+REMOTE_DATA_BUCKET="gs://all-of-us-ehr-dev-peter-speltz"
```
[the remark this answers is not in the document]
Comment (biopete) [reply api/db-cdr/run-bigdata-migrations.sh]: Yes . Put in todo.

File: api/db-cdr/run-bigdata-migrations.sh (hunk 1 above)
Comment (biopete) [reply api/db-cdr/run-bigdata-migrations.sh]: I will check if that was it.

File: api/db-cdr/run-bigdata-migrations.sh (hunk 2 above)
Comment (biopete) [reply api/db-cdr/run-bigdata-migrations.sh]: yep.

**Comment by biopete** [comment]
So annoying i can't find the dang button that submits all comments at once .

File: api/db-cdr/run-bigdata-migrations.sh (hunk 1 above)
Comment (biopete) [reply api/db-cdr/run-bigdata-migrations.sh]: couldn't get this to work.  Just going with curl for now. Something about local permission error on my system writing to ~./config/gcloud but I changed the permissions to allow me to write.

File: api/db-cdr/run-bigdata-migrations.sh (hunk 2 above)
Comment (biopete) [reply api/db-cdr/run-bigdata-migrations.sh]: done

File: api/db-cdr/run-bigdata-migrations.sh [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+echo "Importing data files from $REMOTE_DATA_LOC"
+
+# Add data files to import here
+DATA_FILES=(concept.csv concept_relationship.csv)
+
+# Download data and truncate tables before dropping indexes
+for f in "${DATA_FILES[@]}"
+do
+  local_fpath=/tmp/$f
+  curl -o $local_fpath "$REMOTE_DATA_LOC/$f"
```
Comment (danqrodney) [inline api/db-cdr/run-bigdata-migrations.sh]: Are these files visible publicly? (Does that matter?)

**Comment by biopete** [comment]
Thanks. I wasn't waiting . I just didn't want to merge it until it worked. I will rebase now and test and merge it i guess. Do you want to approve it again ? 

Let's schedule a time to talk about the automated process. I did the cloud sql import manually from the web interface. The data files are public now. I think that is fine as we want them to be public afik. 

I looked a little at gcloud sql instance import 
https://cloud.google.com/sdk/gcloud/reference/sql/instances/import

Looks like that could work eh?

**Pull request merged**